maven-core/src/main/java/org/apache/maven/settings/SettingsUtils.java

```diff
@@ -51,6 +52,25 @@ public static Settings merge(Settings dominant, Settings recessive, String reces
         return new MavenSettingsMerger().merge(dominant, recessive, recessiveSourceLevel);
     }
 
+    /**
+     * @param dominant
+     * @param recessive
+     * @param recessiveSourceLevel
+     * @deprecated please use {@link #merge(Settings, Settings, String)}
+     */
+    @Deprecated
+    public static void merge(
+            org.apache.maven.settings.Settings dominant,
+            org.apache.maven.settings.Settings recessive,
+            String recessiveSourceLevel) {
+
+        if (dominant == null || recessive == null) {
+            return;
+        }
+
+        dominant.delegate = new SettingsMerger().merge(dominant.getDelegate(), recessive.getDelegate(), true, null);
```

Comment:
I described in issue why I use this methods.

`MavenSettingsMerger().merge()` will not work with m-invoker-p because change state of `recessive` - which is reused ...

Reply:
We can suppres this PR  - and work on #874

Requirements that `sourceLevel` can be set once still will be a problem, event if in invoker method which override this filed by reflection can pass by ignore exception. 

Second call on `merge` method will throw exception because of set `sourceLevel` again.